giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/service/delete/impl/DeleteDocumentService.java

```diff
@@ -0,0 +1,151 @@
+    
+    @Autowired
+    private ITransactionalUploadService uploadService;
+    
+    @Autowired
+    private ITransactionalDocumentService documentService;
+    
+    @Autowired
+    private ITransactionalProcessingRequestService processingRequestService;
+    
+    private Map<String, FileType> fileTypes;
+    
+    @PostConstruct
+    public void init() {
+        fileTypes = new HashMap<String, FileType>();
+        
+        // register what content types should be treated as what file type
+        Map<String, IFileTypeHandler> ctxMap = ctx.getBeansOfType(IFileTypeHandler.class);
+        Iterator<Entry<String, IFileTypeHandler>> iter = ctxMap.entrySet().iterator();
+        
+        while(iter.hasNext()){
+            Entry<String, IFileTypeHandler> handlerEntry = iter.next();
+            IFileTypeHandler handler = (IFileTypeHandler) handlerEntry.getValue();
+            for (String type : handler.getHandledFileTypes()) {
+                fileTypes.put(type, handler.getHandledFileType());
+            }
+        }
+        requestFactory.config(StorageDeletionRequest.class);
+    }
+    
+    @Override
+    @Async
+    public void initiateDeletion(IDocument document) throws GilesProcessingException, MessageCreationException {
+        IRequest storageDeletionRequest = createRequest(document);
+        requestProducer.sendRequest(storageDeletionRequest, getTopic());
+    }
+    
+    @Override
+    public void completeDeletion(IDocument document) {
+        deleteFiles(document);
+        deleteProcessingRequests(document);
+        documentService.deleteDocument(document.getId());
+        deleteUpload(document.getUploadId());
+    }
+    
+    private void deleteProcessingRequests(IDocument document) {
+        processingRequestService.deleteRequestsByDocumentId(document.getId());
+    }
+    
+    private void deleteFiles(IDocument document) {
+        fileService.deleteFiles(document.getId());
+    }
+    
+    private void deleteUpload(String uploadId) {
+        // if an upload has multiple documents and only one of the documents is deleted the upload does not have to be deleted.
+        if(documentService.getDocumentsByUploadId(uploadId).isEmpty()) {
+            uploadService.deleteUpload(uploadId);
+        }
+    }
+    
+    private IRequest createRequest(IDocument document) throws GilesProcessingException {
+        IStorageDeletionRequest storageDeletionRequest = null;
+        try {
+            String requestId = documentService.generateRequestId(REQUEST_PREFIX);
+            storeRequestId(requestId, document);
+            storageDeletionRequest = requestFactory.createRequest(requestId, document.getUploadId());
+            storageDeletionRequest.setDocumentId(document.getId());
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new GilesProcessingException(e);
+        } 
+        return storageDeletionRequest;
+    }
+    
+    private String getTopic() {
```

Comment:
this doesn't seem to be used anywhere else than the one call above. And it's a one-liner, so does it really need a method?